packages/hash/src/index.ts

```diff
@@ -2,62 +2,69 @@
 // Inspired by https://github.com/garycourt/murmurhash-js
 // Ported from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash2.cpp#L37-L86
 
+const encoder = new TextEncoder()
+
+// bufferLength must be a multiple of 4 to satisfy Int32Array constraints
+let bufferLength = 2 * 1024
+let buffer = new ArrayBuffer(bufferLength)
+let uint8View = new Uint8Array(buffer)
+let int32View = new Int32Array(buffer)
+
 export default function murmur2(str: string): string {
+  if (str.length > bufferLength) {
+    // buffer.resize() is only available in recent browsers, so we re-allocate
+    // a new buffer and views
+    bufferLength = str.length + (4 - str.length % 4)
+    buffer = new ArrayBuffer(bufferLength)
+
+    uint8View = new Uint8Array(buffer)
+    int32View = new Int32Array(buffer)
+  }
+
+  const length = encoder.encodeInto(str, uint8View).written;
+
   // 'm' and 'r' are mixing constants generated offline.
   // They're not really 'magic', they just happen to work well.
 
-  // const m = 0x5bd1e995;
-  // const r = 24;
+  const m = 0x5bd1e995;
+  const r = 24;
```

Comment:
Inlining constants is not useful, the engine will do it quite early in its optimization passes.